pyathena/sqlalchemy_athena.py

```diff
@@ -372,20 +372,34 @@ def get_schema_names(self, connection, **kw):
                 """
         return [row.schema_name for row in connection.execute(query).fetchall()]
 
-    @reflection.cache
-    def get_table_names(self, connection, schema=None, **kw):
+    def _get_tables(self, connection, schema=None, **kw):
```

Comment:
Looking into the reflection cache implementation, it really is a [basic memoization](https://github.com/sqlalchemy/sqlalchemy/blob/main/lib/sqlalchemy/engine/reflection.py#L43) (capture a function name and its args as key, mapped to the function call result). 
So if we want to make the best (avoid as many Ahtena API calls), I think we should place the here.
```suggestion
    @reflection.cache
    def _get_tables(self, connection, schema=None, **kw):
```
Only remains in the `get_view_names()` and `get_table_names()` functions the filtering of the global list of metadata.
We could cache that also, but the trade-off between memory used and time-gain is less obvious.

Reply:
OK, that looks good.